packages/xsnap/test/test-xsnap.js

```diff
@@ -14,47 +14,100 @@ const xsnapOptions = {
   os: os.type(),
   stderr: 'inherit',
   stdout: 'inherit',
-  debug: true
 };
 
-test('evaluate and issueCommand', async t => {
+function options() {
   const messages = [];
   async function handleCommand(message) {
     messages.push(decoder.decode(message));
     return new Uint8Array();
   }
-  const vat = xsnap({ ...xsnapOptions, handleCommand });
+  return { ...xsnapOptions, handleCommand, messages };
+}
+
+test('evaluate and issueCommand', async t => {
+  const opts = options();
+  const vat = xsnap(opts);
   await vat.evaluate(`issueCommand(ArrayBuffer.fromString("Hello, World!"));`);
   await vat.close();
-  t.deepEqual(['Hello, World!'], messages);
+  t.deepEqual(['Hello, World!'], opts.messages);
 });
 
 test('evaluate until idle', async t => {
-  const messages = [];
-  async function handleCommand(message) {
-    messages.push(decoder.decode(message));
-    return new Uint8Array();
-  }
-  const vat = xsnap({ ...xsnapOptions, handleCommand });
+  const opts = options();
+  const vat = xsnap(opts);
   await vat.evaluate(`
     (async () => {
       issueCommand(ArrayBuffer.fromString("Hello, World!"));
     })();
   `);
   await vat.close();
-  t.deepEqual(['Hello, World!'], messages);
+  t.deepEqual(['Hello, World!'], opts.messages);
+});
+
+test('evaluate and report', async t => {
+  const opts = options();
+  const vat = xsnap(opts);
+  const result = await vat.evaluate(`(() => {
+    const report = {};
+    Promise.resolve('hi').then(v => {
+      report.result = ArrayBuffer.fromString(v);
+    });
+    return report;
```

Comment:
On the one hand, it works and avoids a global. On the other, this is going to surprise every maintainer who comes here, including us in a few months. 🤷‍♀️

Reply:
I started trying to document it but lacked inspiration and then got distracted by lint:types failures.

Reply:
found some inspiration to document it.